### DISCUSSION TITLE
Can you please add a timeout option for the user interface in RTL

### DISCUSSION THREAD

**Discussion by clarketowson** [Ride-The-Lightning/RTL#1339] [Ideas]
Hi Suheb,

I would like to request a new feature for RTL.

It would be great if you could please add an option so a user interface timeout can be set in a future version of RTL.  

Perhaps 5 mins, 15 mins, 30 mins, 60s mins, <user defined value> or unlimited.

This would be great so I can keep RTL open without having to constantly go back into it via BTCPays menu system which is where I access my RTL instance.

Kind Regards,

Clarke

Clarke Towson, BCMS (Bachelor of Computer & Mathematical Science)
CEO
INTJ Billing
 
m: +61 432 359 166
a: 7 Cullen Court Spotswood Victoria 3015 AUSTRALIA
w: https://intjbilling.com/
e: [clarketowson@intjbilling.com](mailto:clarketowson@intjbilling.com)

Bitcoin Lightning Network Node Name: CHILLYCALENDAR 
Node Public Key: 025124c73ef7ecf527e0114ead02a0cc6e3ecbc0c99474ee3f5506c4503b089693 
Status Page: https://intjbilling.com/INTJBillingStatusPage.php 

amboss.space: https://amboss.space/node/025124c73ef7ecf527e0114ead02a0cc6e3ecbc0c99474ee3f5506c4503b089693 
1ml.com: https://1ml.com/node/025124c73ef7ecf527e0114ead02a0cc6e3ecbc0c99474ee3f5506c4503b089693
Lightning network+: https://lightningnetwork.plus/nodes/025124c73ef7ecf527e0114ead02a0cc6e3ecbc0c99474ee3f5506c4503b089693

**Comment by saubyk** [comment]
Hi @clarketowson just so that I am clear, there is a 60 min default timeout currently in RTL and you're requesting flexibility to control the window until the application times out?

**Comment by clarketowson** [comment]
Hi @saubyk.  Yes that's correct.  

Being able to control the timeout would be very useful for me.  I'm running a digital exchange and remittance business in Thailand and I have a number of lightning nodes setup for this purpose and being able to keep the RTL windows open without them timing out enables me to speed up my workflow and more efficiently service the group of people i'm working with when they come to my office to convert their bitcoin to Thai baht.

If this feature could be added that would be amazing!  

To express my appreciation for your work and to help with development I have just made a donation of 150,000 satoshis via your website https://www.ridethelightning.info/ 

Kind Regards,

Clarke

**Reply by saubyk** [reply]
hey Clarke, thanks for the donation. much appreciated.
Created an issue to detail the requirement: https://github.com/Ride-The-Lightning/RTL/issues/1340, will include it in the next release.

Although you should keep in mind that keeping sessions open indefinitely is not a good practice at all from a security perspective. I also added an option of 120 mins, in case that helps your use case.

**Reply by clarketowson** [reply]
Thanks @saubyk you're most welcome.  

I very much appreciate your work and thank you very much for including this functionality in the next release of RTL this is great!

Yes I understand from a security perspective it's not good practice.  I will deal with this aspect via my screen lock on my laptop.  

Very Best Regards,

Clarke
